### DISCUSSION TITLE
allow `_` named fields inside structs to define paddings

### DISCUSSION THREAD

**Discussion by i582** [vlang/v#18636] [Ideas]
### Describe the feature

`_` can be used in multiple places in V and allowing `_` also for field names would make the language more consistent in terms of using the `_` name.

### Use Case

`_` fields can be useful for padding fields:

```v
struct Foo {
	x int
	_ f32  // padding
	y int
}
```

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### Version used

V 0.3.2 89aa695.865c0ea

### Environment details (OS name and version, etc.)

```
OS: macos, macOS, 13.0.1, 22A400
Processor: 10 cpus, 64bit, little endian, Apple M1 Pro
CC version: Apple clang version 14.0.0 (clang-1400.0.29.202)

getwd: /Users/petrmakhnev/vlang-foundation/specification
vmodules: /Users/petrmakhnev/.vmodules
vroot: /Users/petrmakhnev/v
vexe: /Users/petrmakhnev/v/v
vexe mtime: 2023-01-22 21:48:01
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 89aa695.865c0ea

Git version: git version 2.37.1 (Apple Git-137.1)
Git vroot status: weekly.2022.50-266-g865c0ea8-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e
```

**Comment by Delta456** [comment]
What will `println(Foo)` print in this case?

**Comment by spytheman** [comment]
`padding f32` will work just as well, and has the benefit that it already works, also you can use `padding2 u8` etc in the same struct.

Supporting anonymous `_` in struct field declarations, will complicate the compiler for no benefit imho.

**Comment by JalonSolov** [comment]
I understand wanting the anonymous name... mainly so you don't have to have `padding`, `padding2`, `padding3`, ... `padding50`, etc.

**Comment by i582** [comment]
> What will `println(Foo)` print in this case?

Something like this: https://go.dev/play/p/Q8eJTgISXVB

**Comment by i582** [comment]
> `padding f32` will work just as well, and has the benefit that it already works, also you can use `padding2 u8` etc in the same struct.
> 
> Supporting anonymous `_` in struct field declarations, will complicate the compiler for no benefit imho.

As Jalon said, this eliminates the need to name the paddings differently.

Furthermore, this makes the language more consistent, as `_` can be used in more places.

I'm not looking at this from the compiler's point of view, but from the point of view of the language specification, which should describe what the special name `_` is.
